docs/compose-file.md

```diff
@@ -24,6 +24,64 @@ As with `docker run`, options specified in the Dockerfile (e.g., `CMD`,
 `EXPOSE`, `VOLUME`, `ENV`) are respected by default - you don't need to
 specify them again in `docker-compose.yml`.
 
+## Versioning
+
+It is possible to use different versions of the `compose.yml` format.
+Below are the formats currently supported by compose.
```

Comment:
@dnephin Agreed the organization is off.  `## Service configuration reference`  I would most likely put under this heading a paragraph about the fact there are multiple versions.  Then put a cross -reference to the version section to follow.

Then move `## Versioning` to follow the main body of configuration material

This section would need to answer:

* define whether the configuration version matches the product version 
* under what conditions can I use  a different version; for example does the version of the configuration file match or differ from the Compose version
* can I mix syntax in the same config file (prob not but should say)
* can I use two config files that use differing versions of the syntax (I'm thinking also of how the override files interact with the main config under the version conditions)
* provide some guidance to the reader under what circumstances this version mixing is useful

I'll work with the new compose writer @londoncalling to build this suggested organization with @shin- .